### PR TITLE
Respect pod injection annotation for injection analysis

### PR DESCRIPTION
**Pull request by sushicw**
Fixes https://github.com/istio/istio/issues/17771

### CODE REVIEW

File: galley/pkg/config/analysis/analyzers/testdata/injection.yaml [hunk 1]
```diff
@@ -44,3 +44,14 @@ spec:
   containers:
   - image: gcr.io/google-samples/microservices-demo/adservice:v0.1.1
     name: server
+---
+# Pod that explicitly disables injection. Should not generate a warning.
+kind: Pod
+metadata:
+  name: podinjectiondisabled
+  namespace: default
+  sidecar.istio.io/inject: "false"
```
Comment (howardjohn) [inline galley/pkg/config/analysis/analyzers/testdata/injection.yaml]: This isn't right, it should be under `annotations:` I think?

Comment (howardjohn) [reply galley/pkg/config/analysis/analyzers/testdata/injection.yaml]: You need `istioctl analyze` and it would tell you that :slightly_smiling_face:

Comment (sushicw) [reply galley/pkg/config/analysis/analyzers/testdata/injection.yaml]: Hah, whoops! Yeah I thought I'd run the tests successfully before I pushed but I suspect I actually ran the wrong ones. Fixed now.

**Comment by davidebbo** [comment]
@sushicw Is there still time to get this in 1.4?

**Pull request merged**